src/registrar/tests/test_admin.py

```diff
@@ -367,3 +371,83 @@ def test_get_fieldsets_non_superuser(self):
+                        (DomainApplication.submitter.field, 'you'),
+                        (DomainApplication.investigator.field, 'inv'),
+                        (DomainApplication.creator.field, 'cre')
+                        ]
+
+        # Create a sample application - review status does not matter
+        applications = multiple_unalphabetical_applications()
+
+        # Create a mock request
+        request = self.factory.post(
+            "/admin/registrar/domainapplication/{}/change/".format(applications[0].pk)
+        )
+
+        model_admin = AuditedAdmin(DomainApplication, self.site)
+
+        # Typically we wouldn't want two nested for fields,
+        # but both fields are of a fixed length.
+        # For test case purposes, this should be performant.
+        for field in tested_fields:
+            field_name = field[0].name
+            first_name_field = "{}__first_name".format(field_name)
+            last_name_field = "{}__last_name".format(field_name)
+
+            # We want both of these to be lists, as it is richer test wise.
+            # Not really a fan of how this looks, but as the linter demands...
+            desired_order = list(
+                model_admin.get_queryset(request).order_by(
+                                        first_name_field, last_name_field).values_list(
+                                        first_name_field, last_name_field)
+                )
+            current_sort_order: Contact = list(
+                model_admin.formfield_for_foreignkey(field[0], request).queryset
+                )
+
+            # Conforms to the same object structure as desired_order
+            current_sort_order_coerced_type = []
+
+            # This is necessary as .queryset and get_queryset
+            # return lists of different types/structures.
+            # We need to parse this data and coerce them into the same type.
+            for contact in current_sort_order:
+                first = contact.first_name
+                last = contact.last_name
+
+                name_tuple = self.coerced_fk_field_helper(first, last, field[1], ':')
+                if name_tuple:
+                    current_sort_order_coerced_type.append((first, last))
+
+            self.assertEqual(desired_order,
+                             current_sort_order_coerced_type,
+                             "{} is not ordered alphabetically".format(field_name))
+
+    # I originally spent some time trying to fully
+    # generalize this to replace the match/arg fields,
+    # but I think for this specific use case
+    # its not necessary since it'll only be used here.
+    def coerced_fk_field_helper(self, first_name, last_name, field_name, queryset_shorthand): # noqa
+        if(first_name and first_name.split(queryset_shorthand)[1] == field_name): # noqa
+            return (first_name, last_name)
+        else:
+            return None
+
+    def tearDown(self):
+        DomainInformation.objects.all().delete()
```

Comment:
Why is it necessary to clean up DomainINformation?

Reply:
Domains are deleted here? Haven't looked to deep in the code, but if there is an application object and an info object I would assume a domain object exists as well

Reply:
The FK fields that are sorted span across three different domain 'types': info, application, invitation. I am testing on all three